### PR TITLE
added provider parameter on InferenceSession

### PR DESCRIPTION
**Pull request by lazToum**
Add provider parameter on onnxruntime InferenceSession. To avoid the deprecation error:
In ORT 1.9, InferenceSession now requires the providers parameters to be set when enabling Execution Providers other than default CPUExecutionProvider. e.g. InferenceSession('model.onnx', providers=['CUDAExecutionProvider']

### CODE REVIEW

**Comment by dbogunowicz** [comment]
@lazToum are you by chance not using onnxruntime-gpu?

**Comment by lazToum** [comment]
> @lazToum are you by chance using onnxruntime-gpu?

Indeed, I tried the versions 1.10.0 1.11.0 and 1.11.1.

File: src/sparseml/onnx/utils/helpers.py [hunk 1]
```diff
@@ -251,7 +251,11 @@ def extract_nodes_shapes_ort(model: ModelProto) -> Dict[str, List[List[int]]]:
 
     sess_options = onnxruntime.SessionOptions()
     sess_options.log_severity_level = 3
-    sess = onnxruntime.InferenceSession(model_copy.SerializeToString(), sess_options)
+    sess = onnxruntime.InferenceSession(
```
Comment (dbogunowicz) [inline src/sparseml/onnx/utils/helpers.py:254]:
```suggestion
    providers = (
        ["CUDAExecutionProvider", "CPUExecutionProvider"]
        if onnxruntime.get_device() == 'GPU'
        else ["CPUExecutionProvider"]
  
    sess = onnxruntime.InferenceSession(
```

File: src/sparseml/onnx/utils/helpers.py [hunk 2]
```diff
@@ -251,7 +251,11 @@ def extract_nodes_shapes_ort(model: ModelProto) -> Dict[str, List[List[int]]]:
...
+        model_copy.SerializeToString(),
+        sess_options=sess_options,
+        providers=["CPUExecutionProvider"],
```
Comment (dbogunowicz) [inline src/sparseml/onnx/utils/helpers.py:257]:
```suggestion
        providers=providers,
```

File: src/sparseml/onnx/utils/helpers.py (hunk 1 above)
Comment (dbogunowicz) [reply src/sparseml/onnx/utils/helpers.py:254]: Added optional support for onnxruntime-gpu. The point is that the users who installed onnxruntime-gpu and are able to run ort on GPU, do so by default.

**Comment by dbogunowicz** [comment]
@lazToum looking good. I will run some tests and come back to you next week.
I am curious: how did you end up running this function? It is not a frequent pathway in our codebase. How did you ended up encountering an error in there?

**Comment by lazToum** [comment]
> @lazToum looking good. I will run some tests and come back to you next week. I am curious: how did you end up running this function? It is not a frequent pathway in our codebase. How did you end up encountering an error in there?

Great!
To be honest, I started using it as a model compression tool, but I found that it can also provide some useful insights like the flops and parameters counting. I got the ort warning when trying to count a model's total flops.
Great tool btw!

**Review by bfineran — APPROVED** [review]

**Review by markurtz — APPROVED** [review]

**Pull request merged**